renderdoc/driver/shaders/dxbc/dxbc_debug.cpp

```diff
@@ -1861,6 +1861,70 @@ void FlattenVariables(const rdcstr &cbufferName, const rdcarray<ShaderConstant>
   }
 }
 
+void ThreadState::MarkResourceAccess(ShaderDebugState *state, DXBCBytecode::OperandType type,
+                                     const BindingSlot &slot)
+{
+  if(state == NULL)
+    return;
+
+  if(type != DXBCBytecode::TYPE_RESOURCE && type != DXBCBytecode::TYPE_UNORDERED_ACCESS_VIEW)
+    return;
+
+  state->changes.push_back(ShaderVariableChange());
+  ShaderVariableChange &change = state->changes.back();
+  change.after.type = (type == DXBCBytecode::TYPE_RESOURCE) ? VarType::ReadOnlyResource
+                                                            : VarType::ReadWriteResource;
```

Comment:
Minor nit - for various tedious reasons the `makeDebugVariableNode` doesn't give a proper preview for resource values if they don't have `rows` set to 1 since it thinks they're a struct/matrix with children elements. Would make for a nicer experience if we set `change.after.rows = change.after.columns = 1` here. E.g:

![image](https://user-images.githubusercontent.com/661798/80394964-79f2e200-88aa-11ea-84c4-9e8fc2578dc9.png)

vs

![image](https://user-images.githubusercontent.com/661798/80395051-9bec6480-88aa-11ea-9c27-715cf83ba7de.png)